src/elmish-debugger/debugger.fs

```diff
@@ -5,30 +5,45 @@ open Fable.Core
 
 [<RequireQualifiedAccess>]
 module Debugger =
+    open FSharp.Reflection
+
+    let inline private duName (x:'a) = 
+        match FSharpValue.GetUnionFields(x, typeof<'a>) with
+        | case, _ -> case.Name
+
+    let [<PassGenericsAttribute>] private getCase cmd : obj =
+        createObj ["type" ==> duName cmd
+                   "msg" ==> cmd]
+
     type ConnectionOptions =
         | ViaExtension
         | Remote of address:string * port:int
         | Secure of address:string * port:int
 
     let connect =
-        let inline getCase cmd : obj = createObj ["type" ==> unbox cmd?tag
-                                                  "fields" ==> cmd?fields]
+        let serialize = createObj ["replacer" ==> fun _ v -> deflate v]
```

Comment:
I originally tried giving the replacer and reviver fns to this option, but I couldn't get the extractState working when I did that, so I ended up leaving the extract state management alone and just  supplied the replacer.